src/Common/mysqlxx/PoolWithFailover.cpp

```diff
@@ -161,15 +168,15 @@ PoolWithFailover::Entry PoolWithFailover::get()
                     }
 
                     app.logger().warning("Connection to " + pool->getDescription() + " failed: " + e.displayText());
-                    //save all errors to error_detail
-                    if (error_detail.contains(pool->getDescription()))
-                    {
-                        error_detail[pool->getDescription()] = {e.displayText(), e.code()};
-                    }
-                    else
+                    auto [it, inserted] = replica_name_to_error_detail.emplace(pool->getDescription(), ErrorDetail{e.displayText(), e.code()});
+
+                    if (!inserted)
                     {
-                        error_detail.insert({pool->getDescription(), {e.displayText(), e.code()}});
+                        auto & error_detail = it->second;
+                        error_detail.description = e.displayText();
+                        error_detail.code = e.code();
                     }
+
```

Comment:
Maybe just `replica_name_to_error_detail[pool->getDescription()] = ErrorDetail{e.displayText(), e.code()};`?

Reply:
I think `insert_or_assign` is a bit better because it avoids default constructing object if it doesn't exist but the behaviour is the same.